sound/soc/sof/intel/hda-ipc.c

```diff
@@ -264,10 +265,18 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 	}
 
 out:
-	spin_unlock(&sdev->hw_lock);
+	spin_unlock_irq(&sdev->hw_lock);
 	return ret;
 }
 
+bool is_ipc_irq(struct snd_sof_dev *sdev)
+{
+	if (hda_dsp_ipc_irq_handler(0, sdev) == IRQ_WAKE_THREAD)
```

Comment:
can't you remove this function and do all the work here?
Why is a wrapper needed with a first parameter 0 that seems useless?

Reply:
Oh, I just want to reuse the code. Yes, I can remove that function.